funcx_endpoint/funcx_endpoint/endpoint/rabbit_mq/result_queue_publisher.py

```diff
@@ -0,0 +1,86 @@
+import logging
+
+import pika
+
+logger = logging.getLogger(__name__)
+
+
+class ResultQueuePublisher:
+    """ResultPublisher publishes results to a topic EXCHANGE_NAME, with
+    the {endpoint_id}.results as a routing key.
+    """
+
+    EXCHANGE_NAME = "results"
+    EXCHANGE_TYPE = "topic"
+    QUEUE_NAME = "results"
+    GLOBAL_ROUTING_KEY = "*.results"
+
+    def __init__(
+        self,
+        endpoint_id: str,
+        pika_conn_params: pika.connection.Parameters,
+    ):
+        """
+        Parameters
+        ----------
+        endpoint_uuid: str
+            Endpoint UUID string used to identify the endpoint
+        pika_conn_params: pika.connection.Parameters
+            Pika connection parameters to connect to RabbitMQ
+        """
+        self.endpoint_id = endpoint_id
+        self.params = pika_conn_params
+        if self.params._heartbeat is None:
+            self.params._heartbeat = 0
```

Comment:
Why is this `if None -> 0` logic needed?

We're touching a declared-private attribute of a 3rd party library here, so if this is avoidable we should avoid it.

Reply:
For normal operation, I want to disable heartbeats for the result queue. I have this setup to enable setting a heartbeat, and confirm that behavior in tests. If heartbeats are enabled, then this BlockingConnection will timeout without explicit heartbeats setup. In the case of result-publisher where every result message is explicitly acked, heartbeats don't add very much.

Reply:
I don't follow why that means that this class needs to start modifying the parameters it's passed.
The tests need to pass correct parameters and production needs to pass (different) correct parameters.